crates/nu-parser/src/type_check.rs

```diff
@@ -1132,3 +1133,47 @@ fn check_append(
         }
     }
 }
+
+/// If one of the parts of the range isn't a number, a parse error is added to the working set
+pub fn check_range_types(working_set: &mut StateWorkingSet, range: &mut Range, span: Span) {
+    match (&mut range.from, &mut range.next, &mut range.to) {
+        (Some(expr), _, _) | (None, Some(expr), Some(_)) | (None, None, Some(expr))
+            if !type_compatible(&Type::Number, &expr.ty) =>
+        {
+            *expr = Expression::garbage(working_set, expr.span);
+            working_set.error(ParseError::UnsupportedOperationLHS(
+                String::from("range"),
+                span,
+                expr.span,
+                expr.ty.clone(),
+            ));
+        }
+        (Some(lhs), Some(rhs), _) | (Some(lhs), None, Some(rhs)) | (None, Some(lhs), Some(rhs))
+            if !type_compatible(&Type::Number, &rhs.ty) =>
+        {
+            *rhs = Expression::garbage(working_set, rhs.span);
```

Comment:
Sorry I have a question, why do you want to assign `Expression::garbage` to `rhs`?
-------
Regards to this, before assigning `Expression::garbage` to rhs, we should save `rhs.ty` first.  Or else it will show an error: `range is not supported between int and any.`

Which doesn't seem right to me, because `0..(echo 10)` still runs to success, and the rhs type is actually `any`

Reply:
Great catch, I moved all the assignments after the `working_set.error` calls now. The reason for the garbage is to make it be highlighted in red. It's also done in other places (e.g. the math operators) to signal a parse error.